lisa/analysis/idle.py

```diff
@@ -236,6 +281,37 @@ def df_cluster_idle_state_residency(self, cluster):
         idle_time_df.index.name = 'idle_state'
         return idle_time_df
 
+    def _idle_cpus_at(self, timestamp, active_signal_fn):
+        idles = []
+        for cpu in range(self.trace.cpus_count):
+            series_active = active_signal_fn(cpu)
+
+            # Get very first event before 'timestamp'
+            #
+            # XXX: NaN if 'timestamp' is before the first event in the series.
+            # This could be fixed by checking the first event after 'timestamp':
+            # if the CPU goes out of idle later, it was idle during 'timestamp'
+            # and vice-versa.
+            is_active = series_active.reindex([timestamp], method='ffill').iloc[0]
```

Comment:
Can df_adjacent_index() be used here ?

Reply:
Originally I used the both prev/next values, but you suggested using only prev with a reindex which I've just copied here(see previous comments).

Reply:
Right, the original intent of the comment was to avoid dealing with both prev and next. Now that we only care about one of them, either that oneliner or the df_adjacent_index() can be used. df_adjacent_index() seems better since we still have it, as it avoids generating a full intermediate df.